### PR TITLE
Added new blog: Unveiling the Significance of JS ES6 features

### LINKED ISSUE #1578: [Feature Request]: Add new blog: Empowering JavaScript: Unveiling the Significance of ES6
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

I would like to add a new blog article which explains top 10 ES6 features of JS.

### Use Case

In the fast-paced world of web development, staying ahead of the curve is not just an advantage — it’s a necessity. Enter ECMAScript 6, or ES6 for short, a game-changer that has redefined the landscape of JavaScript programming.

### Benefits

_No response_

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by DharshiBalasubramaniyam**
fixes #1578

[Cite any related issue(s) this pull request addresses. If none, simply state "None”]

## Description

I have added a new article which discusses 10 es6 features.

## Type of PR

- [ ] Bug fix
- [x] Feature enhancement
- [ ] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have followed the code style guidelines of this project.
- [x] I have checked for any existing open issues that my pull request may address.
- [x] I have ensured that my changes do not break any existing functionality.
- [x] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [x] I have read the resources for guidance listed below.
- [x] I have followed security best practices in my code changes.

### CODE REVIEW

File: blog/JavaScript ES6 features.md [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+- As we dive into the significance of ES6, we’ll discover how this evolution has not only simplified the developer experience but has also laid the foundation for more robust, expressive, and maintainable code.
+
+- ES6 introduces a plethora of features that elevate JavaScript development to new heights.
+
+
+## 1. let and const
+- In ECMAScript 2015 (ES6), the let and const keywords were introduced to declare variables, offering improvements over the traditional var keyword.
+- Scope: Variables declared with ‘let’ and ‘const’ have block-level scope, meaning they are limited to the block, statement, or expression where ‘var’ has global level scope.
+
+```
```
Comment (Ajay-Dhangar) [inline blog/JavaScript ES6 features.md:23]: replace

````md
```
````

to 

````md
```js
````

for best practice

Comment (Ajay-Dhangar) [reply blog/JavaScript ES6 features.md:23]: and apply for all codes because It is help to view colorful  code

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

File: blog/JavaScript ES6 features.md (hunk 1 above)
Comment (DharshiBalasubramaniyam) [reply blog/JavaScript ES6 features.md]: I have made required changes now.

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**